cmd/operator-sdk/bundle/bundle_suite_test.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2020 The Operator-SDK Authors
```

Comment:
See that the tested added here did not increase the % of coverage:

<img width="763" alt="Screenshot 2020-06-20 at 09 43 30" src="https://user-images.githubusercontent.com/7708031/85197688-8b6ccf00-b2da-11ea-935d-f53fcbba15fa.png">

Also, see that what we really need to cover which are the funcs/cmd logic. See: https://coveralls.io/builds/31574282/source?filename=cmd/operator-sdk/bundle/create.go

In this PR we are checking if the cmd has long and short usages texts as if the default value set in the flag is there which is not required at all.  If we set the usage text and/or a default value using cobra it will be done by the module/cobra code source and not SDK. So, these behaviours are covered/tested by cobra in its repo and we do not need to ensure that. See: https://github.com/spf13/cobra/blob/master/doc/cmd_test.go 

Examples to help you with:


- [bundle/internal/result_test.go](https://github.com/operator-framework/operator-sdk/blob/master/cmd/operator-sdk/bundle/internal/result_test.go) 
- [blog -  How to test CLI commands made with Go and Cobra](https://gianarb.it/blog/golang-mockmania-cli-command-with-cobra) 

IMO: What we need to test?

We need call/execute the command passing the args and then check the expected results. 

Example:

`operator-sdk bundle create`

- Should successfully work when the flag --generated-only is informed and create a bundle in the disk
- Should fail when the crds are not found
- Should fail when is not executed from the root path

PS.: * See [here](https://github.com/operator-framework/operator-sdk/blob/master/internal/generate/olm-catalog/package_manifest_test.go) tests where we create tmp dirs. Note that any tmp data generated with the test need to be cleanup if it faces an issue and or an error.

Reply:
I've already said that I intend to test Run() where the cmd actually does stuff, but am waiting on the kubebuilder integration as it will require a refactor. The test you linked is more like an e2e test as it's testing the actual execution of the cobra command, these are just unit tests testing the cmd layer alone.

Reply:
Hi @jberkhahn, 

Just to clarifies: IMO what was added here is NOT required to be done/tested at all and I'd expected the cmd behaviours to be covered as described in the [blog -  How to test CLI commands made with Go and Cobra](https://gianarb.it/blog/golang-mockmania-cli-command-with-cobra).

Reply:
Really tks for all your help closing it assorted out since we spoke today and agreed in the format to move forward.